india_compliance/gst_india/client_scripts/e_waybill_actions.js

```diff
@@ -106,14 +106,17 @@ function setup_e_waybill_actions(doctype) {
             }
 
             if (
-                frappe.perm.has_perm(frm.doctype, 0, "submit", frm.doc.name) &&
-                can_extend_e_waybill(frm)
+                frappe.perm.has_perm(frm.doctype, 0, "submit", frm.doc.name)
             ) {
-                frm.add_custom_button(
+                const can_extend = can_extend_e_waybill(frm);
+                let btn = frm.add_custom_button(
                     __("Extend Validity"),
-                    () => show_extend_validity_dialog(frm),
+                    can_extend ? () => show_extend_validity_dialog(frm) : null,
                     "e-Waybill"
                 );
+                if(!can_extend) {
+                    btn.addClass('disabled');
```

Comment:
Should this button be visible after the e-Waybill is Invalid and not extendable?